rust/src/utils.rs

```diff
@@ -47,6 +49,29 @@ pub fn symlink<P: ?Sized + nix::NixPath>(path1: &P, path2: &P) -> nix::Result<()
     symlinkat(path1, None, path2)
 }
 
+/// Get mountpoint.
+pub fn get_mountpoint(base_path: &Path) -> Result<&Path, anyhow::Error> {
+    let base_dev = lstat(base_path)?.st_dev;
```

Comment:
On FreeBSD, you can do this much easier by using `statfs` and looking at the `f_mntonname` field.  I'm not sure what the best way to do it on Linux is.

Reply:
Just above, although it isn't very clear.

Reply:
Yeah I choose this way of doing it to ensure that it would work on multiple platforms.